compiler/rustc_mir_transform/src/dest_prop.rs

```diff
@@ -590,13 +576,20 @@ fn save_as_intervals<'tcx>(
             append_at(&mut values, &state, twostep);
             // Ensure we have a non-zero live range even for dead stores. This is done by marking
             // all the written-to locals as live in the second half of the statement.
-            VisitPlacesWith(|place, ctxt| match DefUse::for_place(place, ctxt) {
-                DefUse::Def | DefUse::PartialWrite => {
-                    if let Some(relevant) = relevant.shrink[place.local] {
-                        values.insert(relevant, twostep);
+            let is_simple_assignment =
```

Comment:
This could use an explanatory comment: overlaps aren't allowed so we can only treat the source and destination of an assignment as disjoint if they are "simple" and in a form we can later eliminate at the end of this pass.